src/utils/modul-date/modul-date.ts

```diff
@@ -216,17 +230,57 @@ export default class ModulDate {
         return Math.round(Math.abs(other.innerDate.getTime() - this.innerDate.getTime()) / (1000 * 3600 * 24));
     }
 
+    /**
+     * Add an unit of time to a copy of the current date and return it.
+     *
+     * @param valueToAdd The value to add to the time unit.
+     * @param kind The kind of time unit to be added.
+     * @return A new Date
+     */
+    public add(valueToAdd: number, kind: 'year'): Date {
+        return this.subtract(-valueToAdd, kind);
+    }
+
+    /**
+     * subtract an unit of time to a copy of the current date and return it.
+     *
+     * @param valueToAdd The value to add to the time unit.
+     * @param kind The kind of time unit to be added.
+     * @return A new Date
+     */
+    public subtract(valueToSubstract: number, kind: 'year'): Date {
+        const newDate: Date = new Date(this.innerDate);
+        switch (kind) {
+            case 'year':
+                newDate.setFullYear(newDate.getFullYear() - valueToSubstract);
+                break;
+            default: throw new Error(`modul-date: Unknown substract kind: ${kind}`);
+        }
+
+        return newDate;
+    }
+
+    /**
+     * Return a date representing the end of the day of a given date (23:59:59).
+     *
+     * @return A new Date
+     */
+    public endOfDay(): Date {
+        return new Date(this.innerDate.getUTCFullYear(), this.innerDate.getUTCMonth(), this.innerDate.getUTCDate(), 23, 59, 59, 999);
+    }
+
     private dateFromString(value: string): void {
         if (value === '') {
             this.innerDate = new Date();
         } else {
             this.innerDate = this.convertDateString(value);
         }
-
     }
 
     private convertDateString(value: string): Date {
```

Comment:
C'est rare que je dis ça, mais ça vaudrait probablement la peine d'ajouter des commentaires un peu partout dans cette fonction. Ou des tests unitaires?

Reply:
Ok

Reply:
si on met padString devient static, cette méthode peut-être static aussi, ça serait utile d'éventuellement avoir accès à cette méthode de partout?

Reply:
J'ai renommé pour convertStringToDate et j'ai ajouté des commentaires
// If the string is an iso string, we use it directly.
// Otherwise we try to build the date from a partial date string (2010-12-01 or 2010/12/01)

Reply:
> si on met padString static, cette méthode peut-être static aussi, ça serait utile d'éventuellement avoir accès à cette méthode de partout?

mauvaise idée? je trouve que faire des statics quand c'est possible favorise le principe DRY. Ta méthode n'altère pas et ne divulge pas l'état de ton instance et peut être exposé sans danger. Je veux pas en faire un cas pour cette situation là mais plutôt ouvrir un peu la discussion sur le sujet pour mieux comprendre moi même les best practices.

Reply:
Je comprend pas trop ce que tu veux dire par là.. padString n'est pas static alors je suis un peu clueless =D

Reply:
justement si on met padString static, on peut mettre convertDateString static! :)

Reply:
laisse, pas si important, on en parlera en personne bientôt